tests/drivers/uart/uart_async_api/src/test_uart_async.c

```diff
@@ -736,27 +774,27 @@ ZTEST_USER(uart_async_long_buf, test_long_buffers)
 	memset(long_rx_buf, 0, sizeof(long_rx_buf));
 	memset(long_tx_buf, 1, sizeof(long_tx_buf));
 
-	uart_rx_enable(uart_dev, long_rx_buf, sizeof(long_rx_buf), 10 * USEC_PER_MSEC);
+	UART_RX_ENABLE(uart_dev, long_rx_buf, ARRAY_SIZE(long_rx_buf), 10 * USEC_PER_MSEC);
 
-	uart_tx(uart_dev, long_tx_buf, 500, 200 * USEC_PER_MSEC);
+	expected_nr_rx_bytes = 500 * sizeof(DATA_TYPE);
+	actual_nr_rx_bytes = 0;
+	UART_TX(uart_dev, long_tx_buf, 500, 200 * USEC_PER_MSEC);
+	/* wait for 500 bytes to be received */
 	zassert_equal(k_sem_take(&tx_done, K_MSEC(200)), 0, "TX_DONE timeout");
 	zassert_equal(k_sem_take(&rx_rdy, K_MSEC(200)), 0, "RX_RDY timeout");
-	zassert_equal(long_received[0], 500, "Wrong number of bytes received.");
-	zassert_equal(memcmp(long_tx_buf, long_rx_buf, 500),
```

Comment:
Do we still need to make changes to the callback in UART_RX_RDY in this test? i.e. could we keep callback unchanged and change this part to:
zassert_equal(long_received[0], 500 * sizeof(DATA_TYPE), "Wrong number of bytes received.");
zassert_equal(memcmp(long_tx_buf, long_rx_buf, 500  * sizeof(DATA_TYPE) ),

Reply:
Reverted all my changes